vpn_slice/__main__.py

```diff
@@ -41,19 +41,19 @@ def get_default_providers():
             prep = CheckTunDevProvider,
         )
     elif platform.startswith('darwin'):
-        from distutils.version import LooseVersion
         from platform import release
 
         from .dnspython import DNSPythonProvider
         from .mac import BSDRouteProvider, MacSplitDNSProvider, PfFirewallProvider, PsProvider
         from .posix import PosixHostsFileProvider
+        parsed_release = tuple(int(d) if d.isdigit() else 0 for d in release().split("."))
```

Comment:
What exactly is the format of this `release()` on MacOS?

On Linux, it can look like `6.0.12-123456-generic'` … which would result in a tuple of `(6, 0, 0)`, which means the third component will be entirely lost.

Perhaps not a concern as long as there are always at least two `.` on MacOS.

Reply:
On my machine (macOS Sonoma, 14.3.1):

```shell
$ python3 -c 'import platform; print(platform.release())'
23.3.0
```

Also, looking at this closer I believe the intention from https://github.com/dlenski/vpn-slice/pull/74#issuecomment-803658317 was for this to be compared against `platform.mac_ver()[0]` (which is the macOS version, not the Darwin kernel version):

```shell
$ python3 -c 'import platform; print(platform.mac_ver()[0])'
14.3.1
```

Alteratively, Mac OS X 10.6 was Darwin 10.0 ([ref](https://en.wikipedia.org/wiki/Mac_OS_X_Snow_Leopard#Release_history)) so we could just use `>= (10, 0)` instead. Any preference?

Reply:
Sorry for dropping the ball on this for a while.

> Also, looking at this closer I believe the intention from [#74 (comment)](https://github.com/dlenski/vpn-slice/pull/74#issuecomment-803658317) was for this to be compared against `platform.mac_ver()[0]` (which is the macOS version, not the Darwin kernel version):
> 
> ```shell
> $ python3 -c 'import platform; print(platform.mac_ver()[0])'
> 14.3.1
> ```
> 
> Alteratively, Mac OS X 10.6 was Darwin 10.0 ([ref](https://en.wikipedia.org/wiki/Mac_OS_X_Snow_Leopard#Release_history)) so we could just use `>= (10, 0)` instead. Any preference?

Hmm, no strong preference from me. I don't know the Mac ecosystem well, so I don't know whether this is a more kernel-y thing (use `platform.release`?) or a more userland-y thing (use `platform.mac_ver`).

Reply:
Did it with kernel version `>= (10, 0)` in 4de7bf2.

Reply:
Did it with kernel version >= (10, 0) in https://github.com/dlenski/vpn-slice/commit/4e26adbfd14de2be5e77933e96d353ea7d200107.